misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx

```diff
@@ -22,7 +22,13 @@ const ButtonWithIndicator = ({
 }) => {
   const indicatorClassName = getIndicatorClassName(completeStatus);
 
-  const showHazardsAndAllergens = hazardSymbols != null || allergens != null;
+  const allergensWithColor = allergens != null && allergens.filter((allergen) => allergen.color != null);
+
+  const displayAllergens = allergensWithColor.length > 0;
+
+  const displayHazards = hazardSymbols != null && hazardSymbols.length > 0;
+
+  const showHazardsAndAllergens = displayHazards || displayAllergens;
```

Comment:
rename `showHazardsAndAllergens` to `displayHazardsAndAllergens` (for consistency)